### PR TITLE
fix: web worker repetition and algorithm

### PR DESCRIPTION
**Pull request by amitojsingh**
1. Fixed web worker repetition for the next session.
2. Improved the questions.

### CODE REVIEW

File: src/database/shabadavalidb/words.ts [hunk 1]
```diff
@@ -34,6 +32,23 @@ export const addWordsToSubCollection = async (
   }
 };
 
+export const getWords = async (uid: string, isLearnt: boolean) => {
+  try {
+    const wordsCollectionRef = getWordCollectionRef(uid);
+    const q = query(wordsCollectionRef, where('isLearnt', '==', isLearnt), limit(10));
+    // may add a time condition
+    const querySnapshot = await getDocs(q);
```
Comment (MySmilingTurban) [inline src/database/shabadavalidb/words.ts:40]: add a condition to handle empty response

File: src/pages/dashboard/index.tsx [hunk 2]
```diff
@@ -44,24 +50,32 @@ export default function Dashboard() {
...
+            )}
+          </div>
+        ) : (
           <>
             <Ssa name={user.displayName && userData.displayName} />
             <div className='flex flex-col lg:flex-row text-center justify-center gap-6 h-full h-screen lg:h-3/5 w-full items-center'>
               <WordsSnippetBox commonStyle={commonStyle} />
               <CoinBox commonStyle={commonStyle} />
               <WordBox commonStyle={commonStyle} />
+              {/* <button onClick={() => fetchAlgorithm()}>Test Algorithm </button> */}
```
Comment (MySmilingTurban) [inline src/pages/dashboard/index.tsx:75]: remove comment

File: src/pages/dashboard/index.tsx [hunk 3]
```diff
@@ -24,6 +25,11 @@ export default function Dashboard() {
   const currentLevel: number = useAppSelector((state) => state.currentLevel);
   const currentGamePosition: number = useAppSelector((state) => state.currentGamePosition);
 
+  // const fetchAlgorithm = async () => {
+  //   const { gameArray } = await gameAlgo(user);
+  //   console.log(gameArray);
+  // };
+
```
Comment (MySmilingTurban) [inline src/pages/dashboard/index.tsx:32]: remove comments

File: src/pages/dashboard/index.tsx [hunk 4]
```diff
@@ -11,6 +11,7 @@ import ALL_CONSTANT from 'constants/constant';
 import { useAppSelector } from 'store/hooks';
 import useGamePlay from './hooks/useGamePlay1';
 import Loading from 'components/loading';
+// import { gameAlgo } from './utils';
```
Comment (MySmilingTurban) [inline src/pages/dashboard/index.tsx:14]: remove comment

File: src/pages/dashboard/hooks/useQuestions.ts [hunk 5]
```diff
@@ -1,25 +1,29 @@
...
+import { addQuestionsBatch, getWords } from 'database/shabadavalidb';
+import { createGameScreen, shuffleArray } from '../utils';
 import ALL_CONSTANT from 'constants/constant';
-import { getQuestionsByWordID } from 'database/default';
+import { getQuestions } from 'database/default';
 import { QuestionData } from 'types';
 
-
-const getRandomQuestions = async (user: User, count: number, isLearnt: boolean, questionIds?: string[]) => {
+const getQuestionsHere = async (
```
Comment (MySmilingTurban) [inline src/pages/dashboard/hooks/useQuestions.ts:8]: where is this function being used?
unable to see any files that are using it

File: src/database/default/question.ts [hunk 6]
```diff
@@ -13,9 +13,43 @@ const getOptions = async (wordIDs: string[]) => {
   const options = await Promise.all(optionsPromise);
   return options as Option[];
 };
+const getQuestions = async (wordID: string, questionIDs: string[], needOptions: boolean = true) => {
```
Comment (MySmilingTurban) [inline src/database/default/question.ts:16]: how is it different from getRandomQuestionsByWordID? (apart from the use of randomID comparison in it)
please keep only one function if other is not used anymore

**Review by MySmilingTurban — CHANGES_REQUESTED** [review]

File: src/database/shabadavalidb/words.ts (hunk 1 above)
Comment (amitojsingh) [reply src/database/shabadavalidb/words.ts:40]: done

File: src/pages/dashboard/index.tsx (hunk 2 above)
Comment (amitojsingh) [reply src/pages/dashboard/index.tsx]: done

File: src/pages/dashboard/index.tsx (hunk 3 above)
Comment (amitojsingh) [reply src/pages/dashboard/index.tsx]: done

File: src/database/default/question.ts (hunk 6 above)
Comment (amitojsingh) [reply src/database/default/question.ts:15]: fixed veer ji

**Review by MySmilingTurban — APPROVED** [review]
Looks good to me!

**Pull request merged**